src/ziggurat/new_relic.clj

```diff
@@ -1,6 +1,17 @@
 (ns ziggurat.new-relic
-  (:require [new-reliquary.core :as newrelic]))
+  (:require [new-reliquary.core :as newrelic]
+            [ziggurat.config :refer [ziggurat-config]])
+  (:import (java.util HashMap)
+           [com.newrelic.api.agent NewRelic]))
 
 (defmacro with-tracing [category transaction-name & body]
   `(newrelic/with-newrelic-transaction
      ~category ~transaction-name (^{:once true} fn* [] ~@body)))
+
+(defn- notice-error [throwable message]
+  (NewRelic/noticeError throwable (HashMap. {"error_message" message}) false))
+
+(defn report-error [throwable message]
+  (when (get-in (ziggurat-config) [:new-relic :report-errors])
+    (notice-error throwable message)))
```

Comment:
Can this be done asynchronously? Any degradation in this API call shall not affect the message processing in my opinion. My 2 cents. Need not do it right away. What are your thoughts on the same?

Reply:
I had tried doing it asynchronously but the error being displayed on new relic wasn't the same. The first row is with async, while the second row is sync. Need to debug more on what's happening.
![image](https://user-images.githubusercontent.com/13599208/99184799-8c136f80-276b-11eb-875b-f710964918aa.png)